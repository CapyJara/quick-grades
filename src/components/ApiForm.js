import React from 'react';
import PropTypes from 'prop-types';

const ApiForm = ({ handleChange, handleFormSubmit }) => {
  return (
    <form onSubmit={handleFormSubmit}>
      <input name="apiKey" type="text" placeholder='Enter your api key' onChange={handleChange} />
      <button>Enter key</button>
    </form>
  );
};

ApiForm.propTypes = {
  handleFormSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default ApiForm;
