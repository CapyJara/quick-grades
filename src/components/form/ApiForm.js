import React from 'react';
import PropTypes from 'prop-types';
import styles from './apiForm.css';

const ApiForm = ({ tas, handleChange, handleFormSubmit, getFreshData }) => {
  return (
    <section className={styles.apiForm}>
      <div>

        <form onSubmit={handleFormSubmit} >
          <input
            id='api-input'
            name="apiKey" type="text"
            placeholder='Enter your api key'
            onChange={handleChange} 
            required
          />
          <button>Enter key</button>
        </form>

        <form className={styles.filter}>
          {tas && 
          <>
            <p>Filter By TA</p>
            
            <select 
              id='student-select'
              name='filterTa'
              onChange={handleChange}
            >
              <option></option>
              {tas.map(i => <option key={i}>{i}</option>)}
            </select>
          </>}
        </form>

      </div>

      {tas && <button onClick={getFreshData}>Get Fresh Data</button>}

    </section>
  );
};

ApiForm.propTypes = {
  handleFormSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  getFreshData: PropTypes.func.isRequired,
  tas: PropTypes.oneOfType([PropTypes.bool, PropTypes.array])
};

export default ApiForm;
