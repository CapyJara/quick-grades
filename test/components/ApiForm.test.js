import React from 'react';
import { shallow } from 'enzyme';
import ApiForm from '../../src/components/form/ApiForm';

describe('ApiForm component', () => {
  it('renders ApiForm', () => {
    const wrapper = shallow(<ApiForm 
      tas={[]} 
      handleChange={jest.fn()}
      handleFormSubmit={jest.fn()}
      getFreshData={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
