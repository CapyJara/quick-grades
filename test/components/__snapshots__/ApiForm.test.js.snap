// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiForm component renders ApiForm 1`] = `
<section
  className="apiForm"
>
  <div>
    <form
      onSubmit={[MockFunction]}
    >
      <input
        id="api-input"
        name="apiKey"
        onChange={[MockFunction]}
        placeholder="Enter your api key"
        required={true}
        type="text"
      />
      <button>
        Enter key
      </button>
    </form>
    <form
      className="filter"
    >
      <p>
        Filter By TA
      </p>
      <select
        id="student-select"
        name="filterTa"
        onChange={[MockFunction]}
      >
        <option />
      </select>
    </form>
  </div>
  <button
    onClick={[MockFunction]}
  >
    Get Fresh Data
  </button>
</section>
`;
